### ISSUE TITLE
Subscription status `incomplete` is not supported in types

### ISSUE THREAD

**Issue by spaceemotion** [lmsqueezy/lemonsqueezy.js#83]
I am getting some subscriptions with the status `incomplete` from the API, which is not supported by the typings.

One workaround is to cast it to a `string`, but it's weird that the status does not exist, when the API can return records that have this status:

```ts
if (status.attributes.status as string === 'incomplete') {
  // ...
}
```

This is usually accompanied by even more fields missing data they should have:

```json5
{
	"billing_anchor": 22,
	"cancelled": false,
	"card_brand": null,
	"card_last_four": null,
	"created_at": "2024-01-22T22:32:20.000000Z",
	"customer_id": null,
	"ends_at": null,
	"first_subscription_item": null,
	"order_id": null,
	"order_item_id": null,
	"pause": null,
	"product_id": null,
	"product_name": null,
	"renews_at": "2025-01-22T22:31:24.000000Z",
	"status": "incomplete",
	"status_formatted": "Incomplete",
	"store_id": 0, // redacted, but has store ID
	"test_mode": false,
	"trial_ends_at": null,
	"updated_at": "2024-01-22T22:32:20.000000Z",
	"urls": {
	  // ...
	},
	"user_email": null,
	"user_name": null,
	"variant_id": null,
	"variant_name": null
}
```

**Comment by keyding** [comment]
Hi, @spaceemotion 

I've submitted the fix PR #86.

Also, could you please tell me how you subscribe to a subscription that will have a `status` of `incomplete`, I will update the test case. Thank you.

**Comment by brankoconjic** [comment]
The `incomplete` is included with [v3.1.1](https://github.com/lmsqueezy/lemonsqueezy.js/releases/tag/v3.1.1)

It is a valid status which happens before a subscription has been fulfilled.
